tests/west/runner/test_nrfjprog.py

```diff
@@ -0,0 +1,239 @@
+# Copyright (c) 2018 Foundries.io
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+from unittest.mock import patch, call
+
+import pytest
+
+from west.runner.nrfjprog import NrfJprogBinaryRunner
+from .conftest import RC_KERNEL_HEX
+
+
+#
+# Test values
+#
+
+TEST_DEF_SNR = 'test-default-serial-number'  # for mocking user input
+TEST_OVR_SNR = 'test-override-serial-number'
+
+#
+# Expected results.
+#
+# These record lists of system calls made by the nrfjprog runner. Each
+# system call is a list of arguments passed to the check_call() method.
+#
+# The key naming scheme is <F><SR><SN><E>, where:
+#
+# - F: family, 1 for 'NRF51' or 2 for 'NRF52'
+# - SR: soft reset, Y for yes, N for pin reset
+# - SNR: serial number override, Y for yes, N for 'use default'
+# - E: full chip erase, Y for yes, N for sector / sector and UICR only
+#
+
+EXPECTED_RESULTS = {
+    # NRF51:
+    '1NNN':
+    [['nrfjprog', '--program', RC_KERNEL_HEX, '-f', 'NRF51', '--snr', TEST_DEF_SNR, '--sectorerase'],  # noqa: E501
+     ['nrfjprog', '--pinreset', '-f', 'NRF51', '--snr', TEST_DEF_SNR]],
```

Comment:
Subjective, but tuples instead of lists makes it obvious that these aren't getting modified later (already pretty clear here though...)

Reply:
It was done this way to match the data types provided by the `@patch('...check_call')` decorators below.

In particular, looking here:

```
    assert cc.call_args_list == [call(x) for x in expected]
```

The `call_args_list` attributes use lists:

```
(Pdb) pp cc.call_args_list
[call(['nrfjprog', '--program', 'test-zephyr.hex', '-f', 'NRF51', '--snr', 'test-override-serial-number', '--sectorerase']),
 call(['nrfjprog', '--pinreset', '-f', 'NRF51', '--snr', 'test-override-serial-number'])]
```

We could convert them into tuples of tuples, but why waste cycles?

Reply:
Hmm... should work without conversion though, since a new list is built, and `for x in ...` can iterate through tuples as well as lists.

Not the most important thing in the world though.

Maybe it could be shortened to `assert cc.call_args_list == [call(x) for x in expected_commands(*test_case)]`, since `expected` is only used in one place. Could move `expected` to right before it's used otherwise as well.

Just some personal OCD though...

Reply:
Hmm... thought `call()` would act the same whether you called it with a list or a tuple. Never mind then.

Reply:
`call` is recording a sequence of arguments; the sequence has length 1, and is a list passed to `check_call`. Since `[1,2,3] != (1,2,3)`, I think it's correct for the test to fail when passed a tuple with equal items.

Reply:
Oh, right, the mock just gets the argument list as-is...